Cython/Compiler/ExprNodes.py

```diff
@@ -2661,7 +2661,8 @@ class IteratorNode(ExprNode):
     type = py_object_type
     iter_func_ptr = None
     counter_cname = None
-    cpp_iterator_cname = None
+    cpp_sequence_cname = None
+    cpp_attribute_op = "."
```

Comment:
Is this really needed? Can't the decision be taken locally? Or via a helper method, e.g. `sequence_type.cpp_method_call("begin")` or even `sequence_type.struct_field_code("begin")` or something like that?

Reply:
Not sure. It isn't based on the `sequence_type`, it's based on either `sequence_type` or the temp type, so it'd still needs an attribute to store that. I'm sure it could be done differently, but when I try it typically ends up more convoluted.

Reply:
I think one thing that we can try now that `.analyse_types()` returns a node is to have a dedicated `CppIteratorNode` (sub-)class for C++ iteration, or even have more classes for the different iteration types, maybe with a common internal base class. I think C++ iteration is different enough to merit its own implementation. That would reduce both node classes to their respective use cases and would make these attributes appear less interfering.